### PR TITLE
Adding local ARM compilation/build config

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
This commit allows build NFD operator ARM image locally (either on and dedicated ARM server, or on local x86-64 machine)

### CODE REVIEW

File: Dockerfile.arm [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Build the manager binary
+## FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.11 AS builder
```
Comment (wabouhamad) [inline Dockerfile.arm:2]: Perhaps remove extra comments

File: Dockerfile.arm [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+WORKDIR /go/src/github.com/openshift/cluster-nfd-operator
+
+
+# Build
+COPY . .
+RUN make build
+
+
+# Create production image for running the operator
+## FROM registry.ci.openshift.org/ocp/4.11:base
```
Comment (wabouhamad) [inline Dockerfile.arm:15]: Same here, remove extra comment

**Review by wabouhamad — COMMENTED** [review]
Tested building the cluster-nfd-operator images and was successful and successfully deployed with make deploy after downloading the kustomize binary.  
Other than extra comments, /lgtm

File: Dockerfile.arm (hunk 1 above)
Comment (yevgeny-shnaidman) [reply Dockerfile.arm]: fixed

File: Dockerfile.arm (hunk 2 above)
Comment (yevgeny-shnaidman) [reply Dockerfile.arm]: fixed

File: README.md [hunk 3]
```diff
@@ -65,3 +65,10 @@ https://github.com/kubernetes-sigs/node-feature-discovery#local-user-specific-fe
 The DaemonSet running on the workers will mount the `hostPath: /etc/kubernetes/node-feature-discovery/source.d`. Additional hooks can than be provided by a sidecar container that is as well running on the workers and mounting the same hostpath and writing the hook executable (shell-script, compiled code, ...) to this directory.
 
 NFD will execute any file in this directory, if one needs any configuration for the hook, a separate configuration directory can be created under `/etc/kubernetes/node-feature-discovery/source.d` e.g. `/etc/kubernetes/node-feature-discovery/source.d/own-hook-conf`, NFD will not recurse deeper into the file hierarchy.
+
+## Building NFD operator for ARM locally
+
+There 2 options:
+
+1) Using ARM server - process is the same, but you should use Dockerfile.arm instead of Dockerfile
+2) Using x86 server/laptop - process is the same but before running build command, Makefile.arm should be copied into Makefile
```
Comment (chr15p) [inline README.md:74]: Why do we need a whole separate Makefile for ARM?  can we not just conditionally set the required variables and run it with `make ARCH=arm` for example? Maintaining two separate but parallel makefiles is going to be annoying/painful.

Comment (yevgeny-shnaidman) [reply README.md:74]: NFD is built by ART, so i don't want to interfere with it. Those 2 options are just for local builds, so it seems better to make file separate, and not as options

Comment (wabouhamad) [reply README.md:74]: /lgtm

**Review by wabouhamad — COMMENTED** [review]
/lgtm

**Pull request merged**